File: algorithm/260. Single Number III/260.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        tem=0
        for n in nums:
            tem^=n
        tem=tem&(~(tem-1))
        result=[0,0]
        for n in nums:
            if (tem&n)==0:
                result[0]^=n
            else:
                result[1]^=n
        return result

File: algorithm/260. Single Number III/260.cpp
class Solution {
public:
    vector<int> singleNumber(vector<int>& nums) {
        int tem=0;
        for(int i : nums)
        {
            tem^=i;
        }
        tem=tem&(~(tem-1));
        vector<int> result(2,0);
        for(int i : nums)
        {
            if((i & tem)==0)
                result[0]^=i;
            else
                result[1]^=i;
        }
        return result;
    }
};
